system/Debug/ExceptionHandler.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Debug;
+
+use CodeIgniter\API\ResponseTrait;
+use CodeIgniter\Exceptions\PageNotFoundException;
+use CodeIgniter\HTTP\Exceptions\HTTPException;
+use CodeIgniter\HTTP\IncomingRequest;
+use CodeIgniter\HTTP\RequestInterface;
+use CodeIgniter\HTTP\ResponseInterface;
+use Config\Paths;
+use Throwable;
+
+class ExceptionHandler extends BaseExceptionHandler
+{
+    use ResponseTrait;
+
+    /**
+     * ResponseTrait needs this.
+     */
+    protected ?RequestInterface $request = null;
+
+    /**
+     * ResponseTrait needs this.
+     */
+    protected ?ResponseInterface $response = null;
+
+    /**
+     * Determines the correct way to display the error.
+     *
+     * @return void
```

Comment:
It feels like a big loss not having this in the abstraction. Any reason not to do that?